src/main/kotlin/com/uber/nanoscope/Main.kt

```diff
@@ -219,7 +220,36 @@ class OpenHandler(private val args: List<String>): Runnable {
             exitProcess(1)
         }
 
-        Nanoscope.openTrace(inFile)
+        var sampleFile = File(args[0] + ".timer")
+        var stateFile = File(args[0] + ".state")
+        for ((index, arg) in args.withIndex()){
+            if(index != 0){
+                val parts = arg.split('=')
+                if (parts.size != 2) {
+                    println(usage)
+                    exitProcess(1)
+                }
+
+                if (parts[0] == "--sample-data") {
+                    sampleFile = File(parts[1]);
+                    if (!sampleFile.exists()) {
+                        println("File does not exist: $sampleFile")
+                        exitProcess(1)
+                    }
+                } else if (parts[0] == "--state-data"){
+                    stateFile = File(parts[1]);
+                    if (!stateFile.exists()) {
```

Comment:
This should be factored out to check even if this option isn't used

Reply:
But then the command will fail if someone wants to open an old nanoscope trace because we won't find the .state and .timer log

Reply:
Oh right good point